File: examples/java/kora-java-database-r2dbc/src/main/java/ru/tinkoff/kora/example/r2dbc/R2dbcIdRandomRepository.java
package ru.tinkoff.kora.example.r2dbc;

import java.util.UUID;
import reactor.core.publisher.Mono;
import ru.tinkoff.kora.database.common.UpdateCount;
import ru.tinkoff.kora.database.common.annotation.Column;
import ru.tinkoff.kora.database.common.annotation.Query;
import ru.tinkoff.kora.database.common.annotation.Repository;
import ru.tinkoff.kora.database.r2dbc.R2dbcRepository;

@Repository
public interface R2dbcIdRandomRepository extends R2dbcRepository {

    record Entity(UUID id,
                  @Column("name") String name) {

        public Entity(String name) {
            this(UUID.randomUUID(), name);
        }
    }

    @Query("SELECT * FROM entities_uuid WHERE id = :id")
    Mono<Entity> findById(UUID id);

    @Query("""
            INSERT INTO entities_uuid(id, name)
            VALUES (:entity.id, :entity.name)
            """)
    Mono<UpdateCount> insert(Entity entity);
}

File: examples/kotlin/kora-kotlin-database-r2dbc/src/main/kotlin/ru/tinkoff/kora/kotlin/example/r2dbc/R2dbcIdRandomRepository.kt
package ru.tinkoff.kora.kotlin.example.r2dbc

import ru.tinkoff.kora.database.common.UpdateCount
import ru.tinkoff.kora.database.common.annotation.Column
import ru.tinkoff.kora.database.common.annotation.Query
import ru.tinkoff.kora.database.common.annotation.Repository
import ru.tinkoff.kora.database.r2dbc.R2dbcRepository
import java.util.*

@Repository
interface R2dbcIdRandomRepository : R2dbcRepository {

    data class Entity(val id: UUID = UUID.randomUUID(), @field:Column("name") val name: String)

    @Query("SELECT * FROM entities_uuid WHERE id = :id")
    suspend fun findById(id: UUID): Entity?

    @Query("INSERT INTO entities_uuid(id, name) VALUES (:entity.id, :entity.name)")
    suspend fun insert(entity: Entity): UpdateCount
}


